source/cl/source/extension/source/khr_command_buffer.cpp

```diff
@@ -1072,19 +1072,15 @@ cargo::expected<mux_descriptor_info_s, cl_int> createArgumentDescriptor(
 }  // anonymous namespace
 
 [[nodiscard]] cl_int _cl_command_buffer_khr::updateCommandBuffer(
-    const cl_mutable_base_config_khr &mutable_config) {
+    cargo::array_view<const cl_mutable_dispatch_config_khr *>
+        &mutable_dispatch_configs) {
   const std::lock_guard<std::mutex> guard(mutex);
   const cl_device_id device = command_queue->device;
 
-  const cargo::array_view<const cl_mutable_dispatch_config_khr>
-      mutable_dispatch_configs(mutable_config.mutable_dispatch_list,
-                               mutable_config.num_mutable_dispatch);
-
   // Verify struct configures kernel arguments and return error if malformed
-  for (const auto &config : mutable_dispatch_configs) {
-    OCL_CHECK(config.type != CL_STRUCTURE_TYPE_MUTABLE_DISPATCH_CONFIG_KHR,
-              return CL_INVALID_VALUE);
-
+  for (const auto config_ptr : mutable_dispatch_configs) {
+    OCL_CHECK(config_ptr == nullptr, return CL_INVALID_VALUE);
+    const cl_mutable_dispatch_config_khr config = *config_ptr;
```

Comment:
Originally, `config` was a reference, it should probably remain a reference?

Reply:
I've made this `config` variable a reference and the loop iteration variable `config_ptr`